shared/src/event_handling.rs

```diff
@@ -100,59 +107,66 @@ where
     pub async fn update_events(&mut self) -> Result<()> {
         let range = self.event_block_range().await?;
         tracing::debug!("updating events in block range {:?}", range);
-        let events = self
-            .past_events(&range)
-            .await
-            .context("failed to get past events")?
-            .ready_chunks(INSERT_EVENT_BATCH_SIZE)
-            .map(|chunk| chunk.into_iter().collect::<Result<Vec<_>, _>>());
-        futures::pin_mut!(events);
-        // We intentionally do not go with the obvious approach of deleting old events first and
-        // then inserting new ones. Instead, we make sure that the deletion and the insertion of the
-        // first batch of events happen in one transaction.
-        // This is important for two reasons:
-        // 1. It ensures that we only delete if we really have new events. Otherwise if fetching new
-        //    events from the node fails for whatever reason we might keep deleting events over and
-        //    over without inserting new ones resulting in the database table getting cleared.
-        // 2. It ensures that other users of the database are unlikely to see an inconsistent state
-        //    some events have been deleted but new ones not yet inserted. This is important in case
-        //    for example another part of the code calculates the total executed amount of an order.
-        //    If this happened right after deletion but before insertion, then the result would be
-        //    wrong. In theory this could still happen if the last MAX_REORG_BLOCK_COUNT blocks had
-        //    more than INSERT_TRADE_BATCH_SIZE trade events but this is unlikely.
-        // There alternative solutions for 2. but this one is the most practical. For example, we
-        // could keep all reorg-able events in this struct and only store ones that are older than
-        // MAX_REORG_BLOCK_COUNT in the database but then any code using trade events would have to
-        // go through this class instead of being able to work with the database directly.
-        // Or we could make the batch size unlimited but this runs into problems when we have not
-        // updated it in a long time resulting in many missing events which we would all have to
-        // in one transaction.
-        let mut have_deleted_old_events = false;
-        while let Some(events_chunk) = events.next().await {
-            let unwrapped_events = events_chunk.context("failed to get next chunk of events")?;
-            if !have_deleted_old_events {
-                self.store
-                    .replace_events(unwrapped_events, range.clone())
-                    .await?;
-                have_deleted_old_events = true;
-            } else {
-                self.store.append_events(unwrapped_events).await?;
-            };
+        for (contract, name) in &self.named_contracts {
+            let events = self
```

Comment:
This is the big change here and also what makes it naive. However, I don't think we can get around fetching the blocks twice from within this codebase.